src/main/java/com/arpnetworking/metrics/common/sources/CollectdHttpSource.java

```diff
@@ -0,0 +1,276 @@
+import com.google.common.collect.Maps;
+import net.sf.oval.ConstraintViolation;
+import net.sf.oval.exception.ConstraintsViolatedException;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+/**
+ * Processes HTTP posts from Collectd, Extracts data and emits metrics.
+ *
+ * @author Brandon Arp (brandon dot arp at smartsheet dot com)
+ */
+public class CollectdHttpSource extends ActorSource {
+    @Override
+    protected Props createProps() {
+        return Actor.props(this);
+    }
+
+    /**
+     * Protected constructor.
+     *
+     * @param builder Instance of <code>Builder</code>.
+     */
+    protected CollectdHttpSource(final Builder builder) {
+        super(builder);
+        _parser = builder._parser;
+    }
+
+    private final Parser<List<DefaultRecord.Builder>> _parser;
+
+    /**
+     * Internal actor to process requests.
+     */
+    /* package private */ static final class Actor extends UntypedActor {
+        /**
+         * Creates a {@link Props} for this actor.
+         *
+         * @param source The {@link CollectdHttpSource} to send notifications through.
+         * @return A new {@link Props}
+         */
+        /* package private */ static Props props(final CollectdHttpSource source) {
+            return Props.create(Actor.class, source);
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onReceive(final Object message) throws Exception {
+            if (message instanceof RequestReply) {
+                final RequestReply request = (RequestReply) message;
+                Source.single((HttpRequest) request.getRequest())
+                        .via(_processGraph)
+                        .toMat(_sink, Keep.right())
+                        .run(_materializer)
+                        .whenComplete((done, err) -> {
+                            if (err != null) {
+                                BAD_REQUEST_LOGGER.warn()
+                                        .setMessage("Error handling collectd post")
+                                        .setThrowable(err)
+                                        .log();
+                                if (err instanceof ParsingException) {
+                                    request.getResponse().complete(HttpResponse.create().withStatus(400));
+                                } else if (err instanceof ConstraintsViolatedException) {
+                                    final ConstraintsViolatedException violationException = (ConstraintsViolatedException) err;
+                                    final StringBuilder errorMessages = new StringBuilder();
+                                    for (final ConstraintViolation violation : violationException.getConstraintViolations()) {
```

Comment:
I thought the same thing.  I don't want the field names to become part of the contract, but it's damned nice to see which fields you forgot to set in the HTTP headers.  As for extracting the names of the fields out of the ConstraintViolations, that could be handled in a more formal way.